mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/GenerationContext.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor;
+
+import com.datastax.oss.driver.api.mapper.annotations.Dao;
+import com.squareup.javapoet.ClassName;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.processing.Filer;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+/**
+ * A custom context to share processor-level information with code generators. Basically just a way
+ * to avoid passing a gazillion parameters everywhere.
+ */
+public class GenerationContext {
+
+  private final Map<TypeMirror, EntityDefinition> entityDefinitions = new HashMap<>();
+  private final Map<TypeMirror, ClassName> generatedDaos = new HashMap<>();
```

Comment:
Mmm actually that won't work with incremental compilation: unmodified sources are not passed to the processor again, so we can't expect all DAOs and entities to have been processed in this round.

We'll have to rely on naming conventions to find DAO implementation classes, and re-parse entities on demand (with a cache within the processing round).